Backpack-SwiftUI/Calendar/Classes/Core/CalendarMonthGrid.swift

```diff
@@ -20,110 +20,78 @@ import SwiftUI
-            currentMonthDayCell(numberOfDaysInMonth: numberOfDaysInMonth)
-
-            // Create cells for the days from the next month that are shown in the last week of the current month
-            // The total number of cells used is the sum of the number of days in the current month and the number of
-            // days from the previous month that are shown
-            let totalCellsUsed = numberOfDaysInMonth + daysFromPreviousMonth
-            let remainingCells = daysInAWeek - (totalCellsUsed % daysInAWeek)
+        let grid = calculator.calculateCalendarGrid(monthDate: monthDate)
         
-            remainingEmptyCells(remainingCells: remainingCells)
+        VStack(alignment: .leading) {
+            ForEach(0..<grid.count, id: \.self) { row in
+                weekRow(grid: grid, row: row)
+            }
         }
     }
     
     @ViewBuilder
-    private func previousEmptyCells(daysFromPreviousMonth: Int) -> some View {
-        let preEmptyCells = Array(0..<daysFromPreviousMonth)
-            .map {
-                DayCellIdentifiable(id: "pre-\($0)\(monthDate)", index: $0)
-            }
-        ForEach(preEmptyCells) { _ in
-            VStack(spacing: BPKSpacing.none) {
-                emptyLeadingDayCell()
-                    .frame(height: dayCellHeight)
-                Spacer(minLength: BPKSpacing.none)
+    private func weekRow(grid: [[Date?]], row: Int) -> some View {
+        HStack(alignment: .top, spacing: BPKSpacing.none) {
+            ForEach(0..<grid[0].count, id: \.self) { col in
+                dayCellView(grid: grid, row: row, col: col)
             }
         }
     }
     
     @ViewBuilder
-    private func remainingEmptyCells(remainingCells: Int) -> some View {
-        if remainingCells < daysInAWeek {
-            let remainingEmptyCells = Array(0..<remainingCells)
-                .map {
-                    DayCellIdentifiable(id: "rem-\($0)\(monthDate)", index: $0)
-                }
-            ForEach(remainingEmptyCells) { _ in
-                VStack(spacing: BPKSpacing.none) {
-                    emptyTrailingDayCell()
-                        .frame(height: dayCellHeight)
-                    Spacer(minLength: BPKSpacing.none)
-                }
-            }
+    private func dayCellView(grid: [[Date?]], row: Int, col: Int) -> some View {
+        let dayDate = grid[row][col]
+        if dayDate == nil {
+            emptyCell(grid: grid, row: row)
+        } else if let dayDate {
+            dayCellView(dayDate: dayDate)
         }
     }
     
-    private struct DayCellIdentifiable: Identifiable {
-        let id: String
-        let index: Int
+    @ViewBuilder
+    private func dayCellView(dayDate: Date) -> some View {
+        if !validRange.contains(dayDate) {
+            disabledDayCell(dayDate)
+                .frame(height: dayCellHeight)
+                .frame(maxWidth: .infinity)
+        } else {
+            VStack(spacing: BPKSpacing.sm) {
```

Comment:
the logic for rendering cells is the same as before, not much to look at in this aspect

Reply:
Except is a bit simplified due to not needing a LazyVGrid